src/auxinfo/proof.rs

```diff
@@ -39,6 +39,7 @@ impl AuxInfoProof {
 
     pub(crate) fn prove<R: RngCore + CryptoRng>(
         rng: &mut R,
+        context: &impl ProofContext,
```

Comment:
@marsella is the expectation here that both PiMod and PiFac will have the same context ?

Reply:
Yep. This is the fixed-at-protocol-instantiation context so it will be the same for all proofs in all steps of auxinfo, including pimod and pifac.

Reply:
If you want, you could make this of type `&<AuxInfoParticipant as ProtocolParticipant>::Context` so it's unambiguous. I'm not fully convinced it's nicer, though -- probably better to stick to what you have.